src/Profiles/Shifting/DataPages/Base/ANTPLUS_ShiftingBaseMainDataPageMsg.h

```diff
@@ -0,0 +1,32 @@
+#ifndef ANTPLUS_SHIFTINGBASEMAINDATAPAGEMSG_h
+#define ANTPLUS_SHIFTINGBASEMAINDATAPAGEMSG_h
+
+#include <BaseClasses/ANTPLUS_BaseDataPageMsg.h>
+#include <Profiles/Shifting/ANTPLUS_ShiftingPrivateDefines.h>
+
+#include "ANT.h"
+
+class ShiftingBaseGenericMsg : public BaseDataPageMsg<BroadcastDataMsg> {
+public:
+    ShiftingBaseGenericMsg();
+
+    void copyData(uint8_t * buf, size_t len) { memcpy(_buffer, buf, len); }
```

Comment:
Is this for debug?

Reply:
No, it is the encapsulation for the buffer access (write only access w/o exponation of the buffer pointer).

Reply:
What is the end user case for this? In theory the user can call setData from the the parent class in BroadcastDataMsg to override the data.

Reply:
ShiftingBaseGenericMsg() base class removed